erpnext/stock/doctype/item/item.js

```diff
@@ -785,3 +785,31 @@ frappe.ui.form.on("UOM Conversion Detail", {
 		}
 	}
 })
+
+frappe.ui.form.on("Item Supplier", {
+	supplier: function(frm, cdt, cdn) {
+		let row = locals[cdt][cdn];
+		if (row.supplier) {
+			frappe.call({
+				method: "erpnext.stock.doctype.item.item.get_item_price",
+				args: {
+					supplier: row.supplier,
+					item_code: frm.doc.item_code
+				},
+				callback: function(r) {
+					if (!r.exc && r.message) {
+						frappe.model.set_value(cdt, cdn, "price_list", r.message.price_list);
+						frappe.model.set_value(cdt, cdn, "price_list_rate", r.message.price_list_rate);
+					}
+				}
+			});
+		}
+	},
+	create_purchase_order: function(frm, cdt, cdn) {
+		let row = locals[cdt][cdn];
+		frappe.new_doc("Purchase Order", {
+			supplier: row.supplier,
+			buying_price_list: row.price_list
+		}, true);
```

Comment:
@hrwX  we have item reference too. so its better to add item details in items child table in purchase order.

@neilLasrado this button is in child table of supplier details which add 2 step to create PO. can we bring this button to Item master toolbar let user do multiple PO from dialog box if it has more supplier. @neilLasrado  what do you think?

Reply:
> @hrwX we have item reference too. so its better to add item details in items child table in purchase order.
- This is handled by erpnext, which will add the item details in the child table, so no need to do it maually.

Reply:
@hrwX I test this locally. not able to fetch item details in PO child table.
can you test it again

Reply:
@vishdha Have made a few changes and it should work now, please check